### PR TITLE
Implemented: logic to fetch and update job (#245rk0q)

### PR DESCRIPTION
**Pull request by adityasharma7**

### CODE REVIEW

File: src/components/DurationPopover.vue [hunk 1]
```diff
@@ -1,30 +1,62 @@
 <template>
-  <ion-select :interface-options="customPopoverOptions" interface="popover">
-    <ion-select-option value="a">Every 5 minutes</ion-select-option>
-    <ion-select-option value="b">Every 15 minutes</ion-select-option>
-    <ion-select-option value="c">Hourly</ion-select-option>
-    <ion-select-option value="d">Daily</ion-select-option>
-    <ion-select-option value="e">Disabled</ion-select-option>
+  <!-- TODO Make values dynamic and internationalise text  -->
+  <ion-select :interface-options="customPopoverOptions" interface="popover" :value="getJobStatus(id) ? getJobStatus(id) : 'SERVICE_DRAFT'" @ionChange="updateJob(id)" >
```
Comment (adityasharma7) [inline src/components/DurationPopover.vue:3]: > getJobStatus(id) ? getJobStatus(id) : 'SERVICE_DRAFT' 
 
Handle this in getter?

File: src/components/DurationPopover.vue [hunk 2]
```diff
@@ -1,30 +1,62 @@
...
     IonSelectOption
   },
+  computed: {
+    ...mapGetters({
+      getJobStatus: 'job/getJobStatus',
+      getJob: 'job/getJob'
+    })
+  },
+  methods: {
+    async updateJob(id: string) {
```
Comment (adityasharma7) [inline src/components/DurationPopover.vue:34]: I think we haven't handled the case when user disabled the job. Let me know if I missed something

File: src/components/InventoryPopover.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        payload['tempExprId'] = 'EVERY_15_MIN'
+        payload['jobId'] = job.id
+      }
+
+      job?.status === 'SERVICE_PENDING' ? this.store.dispatch('job/updateJob', payload) : this.store.dispatch('job/scheduleService', payload);
+    }
+  },
+  setup() {
+    const customPopoverOptions: any = {
+      header: 'Schedule inventory hard sync',
```
Comment (adityasharma7) [inline src/components/InventoryPopover.vue:51]: I think this needs to be internationalised

File: src/store/modules/job/JobState.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default interface JobState {
+    preorder: any;
```
Comment (adityasharma7) [inline src/store/modules/job/JobState.ts:2]: We could remove if unused

File: src/store/modules/job/actions.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          status: item.statusId
+        }
+      })
+
+      commit(types.JOB_UPDATED_BULK, cached);
+    }
+    return resp;
+  },
+  async updateJob ({ commit }, payload) {
+    // const resp = await JobService.updateJob(payload)
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:48]: It seems we left code commented

File: src/store/modules/job/actions.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    // if (resp.status === 200 && !hasError(resp) && resp.data.docs) {
+    //   commit(types.JOB_UPDATED, { job: payload});
+    // }
+    // return resp;
+    console.log('job updated', payload)
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:53]: Missed console log

File: src/store/modules/job/actions.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  },
+
+  async scheduleService({ commit }, payload) {
+    // const resp = await JobService.scheduleJob(payload);
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:57]: commented code with console log

File: src/store/modules/job/getters.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { GetterTree } from 'vuex'
+import JobState from './JobState'
+import RootState from '../../RootState'
+
+const getters: GetterTree <JobState, RootState> = {
+    getPreOrderInformation (state) {
```
Comment (adityasharma7) [inline src/store/modules/job/getters.ts:6]: We could remove unused getters

File: src/store/modules/job/mutations.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import { MutationTree } from 'vuex'
+import JobState from './JobState'
+import * as types from './mutation-types'
+
+const mutations: MutationTree <JobState> = {
+    [types.JOB_UPDATED_BULK] (state, payload) {
+        state.cached = payload
+        console.log(state.cached)
```
Comment (adityasharma7) [inline src/store/modules/job/mutations.ts:8]: Console log missed

File: src/store/modules/job/actions.ts (hunk 5 above)
Comment (ymaheshwari1) [reply src/store/modules/job/actions.ts:48]: Sir, commented this for testing, will uncomment this.

File: src/components/DurationPopover.vue (hunk 1 above)
Comment (ymaheshwari1) [reply src/components/DurationPopover.vue]: Sir, handled this in getter to return default value.

File: src/store/modules/job/JobState.ts (hunk 4 above)
Comment (ymaheshwari1) [reply src/store/modules/job/JobState.ts]: Removed unused state.

File: src/store/modules/job/actions.ts (hunk 5 above)
Comment (ymaheshwari1) [reply src/store/modules/job/actions.ts]: Done Sir

File: src/store/modules/job/actions.ts (hunk 6 above)
Comment (ymaheshwari1) [reply src/store/modules/job/actions.ts]: Removed console log

File: src/store/modules/job/getters.ts (hunk 8 above)
Comment (ymaheshwari1) [reply src/store/modules/job/getters.ts]: Removed unused getters.

File: src/store/modules/job/mutations.ts (hunk 9 above)
Comment (ymaheshwari1) [reply src/store/modules/job/mutations.ts]: Removed console log statement

File: src/components/InventoryPopover.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/components/InventoryPopover.vue]: Added text for internationalization.

File: src/store/modules/job/actions.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        "statusId_op": "in",
+        ...payload.inputFields
+      },
+      "entityName": "JobSandbox",
+      "noConditionFind": "Y"
+    })
+    if (resp.status === 200 && !hasError(resp) && resp.data.docs) {
+      const cached = JSON.parse(JSON.stringify(state.cached));
+
+      resp.data.docs.filter((job: any) => job.statusId === 'SERVICE_PENDING').map((item: any) => {
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts]:
```suggestion
      resp.data.docs.filter((job: any) => job.statusId === 'SERVICE_PENDING').map((job: any) => {
```

File: src/store/modules/job/actions.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        return cached[item.systemJobEnumId] = {
+          ...item,
+          id: item.jobId,
+          frequency: item.tempExprId,
+          enumId: item.systemJobEnumId,
+          status: item.statusId
+        }
+      })  
+
+      resp.data.docs.filter((job: any) => job.statusId === 'SERVICE_DRAFT').map((item: any) => {
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts]:
```suggestion
      resp.data.docs.filter((job: any) => job.statusId === 'SERVICE_DRAFT').map((job: any) => {
```

**Pull request merged**